File: credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    long n;
    do
    {
        n = get_long("Number:" );
    }
    while (n<4000000000000 || n>5500000000000000);
   
    long u=n;
    int a=0;
    do
    {
            u=u/1e1;
            a++;
    }
    while((u)>1);
   
   int p;
   int k=2;
   int l=1;
   int x;
   int x2;
   int y;
   int y2;
   int sum = 0;
   int sum2 =0;
   int k1;

  
    k1=(k/2)%10;
    if (k1==0)
    {
        do
        {
            x=n%10^l;
            sum=+x;
            l=l+2;
        }
        while(l<=a);
        do
        {
            y=n%10^k;
            y2=y*2;
            if(y2>10)
            {
                y2=y2%10+(int)y2/10;
                sum2=+y2;
                l=k+2;
            }
            else
            {
                sum2=+y2;
                l=k+2;
            }
        }
        while(k<=a);
    }
    else
     {
        do
        {
            x=n%10^k;
            sum=+x;
            l=k+2;
        }
        while(k<=a);
        do
        {
            y=n%10^l;
            y2=y*2;
            if(y2>10)
            {
                y2=y2%10+(int)y2/10;
                sum2=+y2;
                l=l+2;
            }
            else
            {
                sum2=+y2;
                l=l+2;
            }
        }
        while(l<=a);
    }
   printf("%i,%i", sum,sum2);
}

File: credit.py
import cs50

def main():
    card = cs50.get_string("Number: ")
    if(valid(card) == False or len(card) < 13):
         print("INVALID")
    else:
        print(company(card))
    
def valid(card):
    suma1 = 0
    suma2 = 0
    k = 0
    if (len(card) % 2 == 0):
        for i in range(0, len(card) -1 , 2): 
            k = (int(card[i]) * 2)
            if k>9:
                suma1 = suma1 + int(k/10) + int(k%10)
            else:
                suma1 += k
            
          
        for i in range(1, len(card) , 2):
            suma2 += int(card[i])
    else:
        for i in range(1, len(card)-1 , 2): 
            k = (int(card[i]) * 2)
           
            if k>9:
                suma1 = suma1 + int(k/10) + int(k%10)
            else:
                suma1 += k
            
          
        for i in range(0, len(card) , 2):
            suma2 += int(card[i])
            
    suma = suma1 + suma2
    
    return (suma % 10 == 0)

def company(card):
    temp = []
    temp.append(int(card[0]))
    temp.append(int(card[1]))
    
        
    if (temp[0] == 5 and temp[1] in range(1,6)):
        return "MASTERCARD"
    if (temp[0] == 3 and temp[1] in [4, 7]):
        return "AMEX"
    if (temp[0] == 4):
        return "VISA"
        
main()
    